### PR TITLE
ASHA facilitators custom report

### PR DESCRIPTION
**Pull request by snopoke**
A re-work of the CRS ASHA facilitators report after clarifications in the spec revealed problems with the current report.

Also making use of new UCR custom data source.

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: custom/up_nrhm/data_sources/asha_facilitators.json [hunk 1]
```diff
@@ -0,0 +1,170 @@
+{
```
Comment (snopoke) [inline custom/up_nrhm/data_sources/asha_facilitators.json]: @kkrampa we're wanting to use these new data source configurations to replace fluff pillows. There is documentation on them here: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/userreports/README.md

**Comment by czue** [comment]
build fail real?
```
Traceback (most recent call last):
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/management/base.py", line 222, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/management/commands/test.py", line 72, in execute
    super(Command, self).execute(*args, **options)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/management/base.py", line 255, in execute
    output = self.handle(*args, **options)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/south/management/commands/test.py", line 8, in handle
    super(Command, self).handle(*args, **kwargs)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/management/commands/test.py", line 89, in handle
    failures = test_runner.run_tests(test_labels)
  File "/home/travis/build/dimagi/commcare-hq/testrunner.py", line 259, in run_tests
    all_suite = self.build_suite(all_test_labels, extra_tests)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/test/simple.py", line 256, in build_suite
    app = get_app(label)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/db/models/loading.py", line 157, in get_app
    raise ImproperlyConfigured("App with label %s is missing a models.py module." % app_label)
ImproperlyConfigured: App with label up_nrhm is missing a models.py module.
```

File: custom/up_nrhm/models.py [hunk 2]
```diff
@@ -1,53 +0,0 @@
-from functools import partial
```
Comment (czue) [inline custom/up_nrhm/models.py]: it looks like you need to keep this file around to make the build happy

Comment (czue) [inline custom/up_nrhm/models.py]: (just the file, not the contents)

File: custom/up_nrhm/sql_data.py [hunk 3]
```diff
@@ -1,13 +1,127 @@
...
+                from
+                    asha_facilitators
+                where
+                    date between '2015-01-01' and '2015-02-01'
+                    and owner_id = 'abcd'
+                group by case_id
+            ) as max
+        where
+            orig.case_id = max.case_id
+            and orig.date = max.date
```
Comment (czue) [inline custom/up_nrhm/sql_data.py]: pretty cool - i didn't know you could do this in the framework

File: settings.py [hunk 4]
```diff
@@ -1221,8 +1221,6 @@
         'custom.tdh.models.TDHNewbornTreatmentFluffPillow',
         'custom.tdh.models.TDHChildClassificationFluffPillow',
         'custom.tdh.models.TDHChildTreatmentFluffPillow',
-        'custom.up_nrhm.models.UpNRHMLocationHierarchyFluffPillow',
-        'custom.up_nrhm.models.ASHAFacilitatorsFluffPillow'
```
Comment (czue) [inline settings.py]: :hammer: :bomb: :boom: all the pillows!

**Comment by czue** [comment]
looks good - will wait for test fixes

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by snopoke** [comment]
@czue :white_check_mark:

File: custom/up_nrhm/data_sources/asha_facilitators.json (hunk 1 above)
Comment (kkrampa) [inline custom/up_nrhm/data_sources/asha_facilitators.json]: @snopoke Run load_spec command causes KeyError: 'doc_type'. Adding doc_type causes another error BadValueError: Property domain is required. Should I do something before load_spec or add some param?

Comment (snopoke) [inline custom/up_nrhm/data_sources/asha_facilitators.json]: You don't need to use `load_spec` for these. They are auto-loaded via a setting: https://github.com/dimagi/commcare-hq/pull/5535/files#diff-bdf3ecebd8379ca98cc89e545fc90899R1238

Here is the class that loads them: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/userreports/models.py#L267-267

Steps to test locally:
* Run CommCare HQ and enable the 'User configurable reports UI' feature flag (http://localhost:8000/hq/flags/edit/user_reports/)
* Navigate to: http://localhost:8000/a/up-nrhm/configurable_reports/
* Select the datasource ("ASHA Facilitators")
* Click on the "Rebuild" button at the bottom of the form (celery must be running)

If you want to enable continuous updates to the data you need to run the `CustomDataSourcePillow`:

    ./manage.py run_ptop --pillow-name=CustomDataSourcePillow

**Pull request merged**